src/sentry/api/endpoints/user_details.py

```diff
@@ -168,6 +165,18 @@ def put(self, request: Request, user) -> Response:
         if not serializer.is_valid() or not serializer_options.is_valid():
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+        if not request.access.has_permission("users.admin"):
+            if not user.is_superuser and request.data.get("isSuperuser"):
+                return Response(
+                    {"detail": "Missing required permission to add superuser."},
+                    status=status.HTTP_403_FORBIDDEN,
+                )
+            elif not user.is_staff and request.data.get("isStaff"):
+                return Response(
+                    {"detail": "Missing required permission to add admin."},
+                    status=status.HTTP_403_FORBIDDEN,
+                )
+
```

Comment:
nit: Permission checks usually go before any serialization. Either that, or you can add `validate_is_staff()` and `validate_is_superuser()` methods to the `BaseUserSerializer` and just rely on that for throwing errors.